src/v0/destinations/mp/transform.js

```diff
@@ -226,17 +230,92 @@ const processTrack = (message, destination) => {
   return returnValue;
 };
 
+function trimTraits(traits, contextTraits, setOnceProperties) {
+  // Create a copy of the original traits object
+  const traitsCopy = { ...traits };
+  const contextTraitsCopy = { ...contextTraits };
+
+  // Initialize setOnce object
+  const setOnceEligible = {};
+
+  // Step 1: find the k-v pairs of setOnceProperties in traits and contextTraits
+
+  setOnceProperties.forEach((propertyPath) => {
+    const pathSegments = propertyPath.split('.');
+    const propName = pathSegments[pathSegments.length - 1];
+
+    if (Object.keys(traitsCopy).length > 0 && get(traitsCopy, propertyPath)) {
+      setOnceEligible[propName] = get(traitsCopy, propertyPath);
```

Comment:
Assign get(traitsCopy, propertyPath) to some variable instead of calling 2 times.

Reply:
done